services/spar/test/Test/Spar/Sem/ScimExternalIdStoreSpec.hs

```diff
@@ -1,192 +1,16 @@
-{-# LANGUAGE QuantifiedConstraints #-}
-{-# OPTIONS_GHC -Wno-orphans #-}
 {-# OPTIONS_GHC -fplugin=Polysemy.Plugin #-}
 
 module Test.Spar.Sem.ScimExternalIdStoreSpec where
 
 import Arbitrary ()
-import Data.Id
 import Imports
 import Polysemy
-import Polysemy.Check
-import qualified Spar.Sem.ScimExternalIdStore as E
 import Spar.Sem.ScimExternalIdStore.Mem
+import Spar.Sem.ScimExternalIdStore.Spec
 import Test.Hspec
 import Test.Hspec.QuickCheck
-import Test.QuickCheck
-
-deriveGenericK ''E.ScimExternalIdStore
-
-propsForInterpreter ::
-  PropConstraints r f =>
-  String ->
-  (forall a. f a -> a) ->
-  (forall a. Sem r a -> IO (f a)) ->
-  Spec
-propsForInterpreter interpreter extract lower = do
-  describe interpreter $ do
-    prop "delete/delete" $ prop_deleteDelete Nothing lower
-    prop "delete/lookup" $ prop_deleteLookup (Just $ show . (() <$) . extract) lower
-    prop "delete/insert" $ prop_deleteInsert Nothing lower
-    prop "lookup/insert" $ prop_lookupInsert Nothing lower
-    prop "insert/delete" $ prop_insertDelete Nothing lower
-    prop "insert/lookup" $ prop_insertLookup (Just $ show . (() <$) . extract) lower
-    prop "insert/insert" $ prop_insertInsert (Just $ show . (() <$) . extract) lower
 
 spec :: Spec
 spec = modifyMaxSuccess (const 1000) $ do
   propsForInterpreter "scimExternalIdStoreToMem" snd $ pure . run . scimExternalIdStoreToMem
```

Comment:
same here.

but i realize now that i haven't though this through.  i know we have something close to a proof that effects don't interfere with each other, but i guess i always thought that we have two interpreters that handle all the effects under `Spar.Sem`: one for production code and one for the unit tests; and that we're running all the tests, including these here, the unit tests covering the application logic, and the intergration tests checking for equivalence between the two interpreters.

does this idea even make sense?  is it reasonable?  would it be more expensive than hand-picking interpreters for just the one effect you're dealing with in every particular module?

Reply:
I'm not sure I understand the question. My plan is to still test the entire application via test interpreters, but that only finds bugs exercised by the existing test suite. These property tests should give us a good idea if there are bugs hiding in the existing effects themselves.